### PR TITLE
common: fix privatefile create. 

### PR DESCRIPTION
**Pull request by 0xFFFC0000**
When creating a private file we need to delete the file if exists.

### CODE REVIEW

**Review by vtnerd — APPROVED** [review]

**Comment by selsta** [comment]
@0xFFFC0000 please also open against release branch

File: src/common/util.cpp [hunk 1]
```diff
@@ -123,6 +123,14 @@ namespace tools
 
   private_file private_file::create(std::string name)
   {
+    if (epee::file_io_utils::is_file_exist(name)) {
```
Comment (iamamyth) [inline src/common/util.cpp:126]: This change seems structurally and semantically quite dubious:
1. Create tends to be a one-way operation, very different from drop and recreate. If you want to support a "replace" operation, you'd be better off making a new function which wraps the existing and allow the caller here to optionally request O_EXCL.
2. The existing usage of this function that I see in wallet_rpc_server seems well aware it *doesn't* remove the existing file; what caller does this help, or contract (documented in the comments or implied by usage) does it honor?

Relevant wallet_rpc_server code:
```
        rpc_login_file = tools::private_file::create(temp);
        if (!rpc_login_file.handle())
        {
          LOG_ERROR(tr("Failed to create file ") << temp << tr(". Check permissions or remove file"));
          return false;
        }
```

Comment (0xFFFC0000) [reply src/common/util.cpp]: Thanks for your comment @iamamyth 
Few points:

1. Apologies for the late reply, I have been extremely busy the past few weeks.
2. I thought about your comment. And your criticism makes sense to me.

In the new push, I am addressing those issues you have raised.

**Review by selsta — APPROVED** [review]

File: src/common/util.cpp (hunk 1 above)
Comment (iamamyth) [reply src/common/util.cpp]: Looks good; thanks for spotting and fixing the bug.

**Review by iamamyth — APPROVED** [review]

**Pull request merged**